executor.go

```diff
@@ -0,0 +1,284 @@
+package main
```

Comment:
This file is code that mostly used to live in `main.go` and a couple of other files.